pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -293,6 +294,24 @@ func (adc *attachDetachController) nodeAdd(obj interface{}) {
 	// the attached volumes field. This function ensures that we sync with
 	// the actual status.
 	adc.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
+
+	if _, exists := node.Annotations[volumehelper.ControllerManagedAttachAnnotation]; exists {
```

Comment:
Actually I think waitForCacheSync in #run will help here. It will prevent the reconciler from running until all the nodes/pods have been processed. Still a small race since the handlers operate asynchronously so may still be running even when waitForCacheSync returns.